### PR TITLE
Make tests python3 compatiable

### PR DESCRIPTION
**Pull request by Cjkjvfnby**
I have started form investigation about ReadOnlyDicts and end up using `six` and make all test compatible for python2 and python3.

I have updated Travis build to run tests on all pythons. I have added only least and last used pythons. 

AI code is not yet fully compatible, only partially covered by tests.

- `test_str_conversion` was removed. It depends on the dictionary iteration order, which is unordered by the language specification. Python 2.7 and 3.5  have different implementation for a dictionary so it fails in 3.5 and passes in 2.7.  In python 3.7 dictionary implementation was changed again, now iteration order is guaranteed.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -136,7 +136,10 @@ jobs:
       os: linux
       dist: xenial
       language: python
-      python: 2.7.14
+      python:
```
Comment (Cjkjvfnby) [inline .travis.yml]: Looks like test was run only for python 2.7.14

File: .travis.yml [hunk 2]
```diff
@@ -136,7 +136,10 @@ jobs:
...
+        - "2.7.14"
+        - "3.6"
```
Comment (Cjkjvfnby) [inline .travis.yml]: Skip 3.5 since it may require additional coniguration
https://stackoverflow.com/a/41262275

**Comment by o01eg** [comment]
I suppose you should do rebase over `master` instead of merge.

**Comment by Cjkjvfnby** [comment]
> I suppose you should do rebase over master instead of merge.

I plan to squash all changes into a single commit.  Rebase time has not come yet.

**Comment by Cjkjvfnby** [comment]
I am finished with this PR. 

It is ready for review.

**Review by o01eg — APPROVED** [review]

**Comment by Morlic-fo** [comment]
@geoffthemedio This introduces a (MIT) licensed third party file. Does this require further changes/mentions?

**Comment by geoffthemedio** [comment]
Based on a quick google, MIT licensed files can be re-licensed as GPL, as long as the copyright notice is left, and it's there in the commit... so I think it's fine.

**Pull request merged**